### PR TITLE
Migrate TestAddonTitle.js, TestHeadLinks.js and TestHeadMetaTags.js

### LINKED ISSUE #11166: Migrate /components/TestAddonTitle.js to react-testing-library
Migrate /components/TestAddonTitle.js to react-testing-library

### LINKED ISSUE #11204: Migrate /components/TestHeadLinks.js to react-testing-library
Migrate /components/TestHeadLinks.js to react-testing-library

### LINKED ISSUE #11205: Migrate /components/TestHeadMetaTags.js to react-testing-library
Migrate /components/TestHeadMetaTags.js to react-testing-library

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #11166 
Fixes #11204 
Fixes #11205 

I am working on `pages/TestAddon.js` and identified these as components that are used in `Page`, but also elsewhere, so it makes sense for them to have their own unit tests.

### CODE REVIEW

File: tests/unit/helpers.js [hunk 1]
```diff
@@ -1558,9 +1559,35 @@ export const screen = {
   ...customQueries,
 };
 
+const getByTextInDocument = (text) => {
```
Comment (eviljeff) [inline tests/unit/helpers.js:1562]: can you explain what this function is for, contrasting to `getByText`

**Review by eviljeff — APPROVED** [review]
tbh not really sure what the `getElement` and `getElements` functions are giving us over the underlying DOM query function, beyond making us feel better about not having to stick eslint-enable testing-library/no-node-access around each test that uses them😬

**Comment by bobsilverberg** [comment]
> tbh not really sure what the `getElement` and `getElements` functions are giving us over the underlying DOM query function, beyond making us feel better about not having to stick eslint-enable testing-library/no-node-access around each test that uses them😬

`getElements` is shorter than `document.querySelectorAll`, and `getElement` is shorter than `document.querySelectorAll(x)[0]`. Also, it allows us to change the implementation if we ever decide to do so. And, as you point out, we don't have to continually add the `eslint` comment. I do not think we should use these things often, but it seems that `screen` has no way of detecting these tags, so, for specific tests where we care about things like `link` and `meta` tags, we need some way of finding them.

File: tests/unit/amo/components/TestAddonTitle.js [hunk 2]
```diff
@@ -156,74 +140,76 @@ describe(__filename, () => {
...
-    // Finally, it should be the "by"
-    expect(authors.childAt(5).text()).toEqual('by');
+    expect(screen.getAllByRole('link')).toHaveLength(2);
+    const link1 = screen.getByRole('link', { name: author1.name });
+    expect(link1).toHaveAttribute('href', `/fa/android/user/${author1.id}/`);
+    const link2 = screen.getByRole('link', { name: author2.name });
+    expect(link2).toHaveAttribute('href', `/fa/android/user/${author2.id}/`);
+
+    expect(screen.getByClassName('AddonTitle-author')).toHaveTextContent(
+      `${author1.name} ,${author2.name} by`,
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestAddonTitle.js:157]: Because it is RTL (real RTL this time, not `react-testing-library`). This matches the previous assertion (although the previous one is a bit hard to parse because of how it is broken up into so many pieces).

File: tests/unit/amo/components/TestHeadLinks.js [hunk 3]
```diff
@@ -1,58 +1,70 @@
...
+      store,
+    };
+
+    return defaultRender(<HeadLinks {...props} />, renderOptions);
   };
 
   it.each([CLIENT_APP_ANDROID, CLIENT_APP_FIREFOX])(
     'renders alternate links with hreflang for %s',
-    (clientApp) => {
+    async (clientApp) => {
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestHeadLinks.js:35]: It allows us to use `await`, which we need for `waitFor`, which we need to wait for the changes to the `head` to be rendered. Other than that it shouldn't have an effect on the tests.

File: tests/unit/amo/components/TestAddonTitle.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eviljeff) [reply tests/unit/amo/components/TestAddonTitle.js:157]: oh, okay, I missed the test string mentioning RTL

File: tests/unit/helpers.js (hunk 1 above)
Comment (eviljeff) [reply tests/unit/helpers.js:1562]: I think that's a better name, yes.  (I thought of `getByFlatText` but I'm not convinced its better 🤷 )
